HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 1s [Retry 1/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 2s [Retry 2/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 4s [Retry 3/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 8s [Retry 4/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 8s [Retry 5/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 1s [Retry 1/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 2s [Retry 2/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 4s [Retry 3/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 8s [Retry 4/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 8s [Retry 5/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Using the latest cached version of the dataset since flwrlabs/pacs couldn't be found on the Hugging Face Hub
Found the latest cached dataset configuration 'default' at /home/apiasecz/.cache/huggingface/datasets/flwrlabs___pacs/default/0.0.0/394113073258ead631f617d2e13bb377c0715c4b (last modified on Wed Jan 29 16:59:35 2025).
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 1s [Retry 1/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 2s [Retry 2/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 4s [Retry 3/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 8s [Retry 4/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 8s [Retry 5/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 1s [Retry 1/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 2s [Retry 2/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 4s [Retry 3/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 8s [Retry 4/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 8s [Retry 5/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Using the latest cached version of the dataset since flwrlabs/pacs couldn't be found on the Hugging Face Hub
Found the latest cached dataset configuration 'default' at /home/apiasecz/.cache/huggingface/datasets/flwrlabs___pacs/default/0.0.0/394113073258ead631f617d2e13bb377c0715c4b (last modified on Wed Jan 29 16:59:35 2025).
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 1s [Retry 1/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 2s [Retry 2/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 4s [Retry 3/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 8s [Retry 4/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 8s [Retry 5/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 1s [Retry 1/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 2s [Retry 2/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 4s [Retry 3/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 8s [Retry 4/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 8s [Retry 5/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Using the latest cached version of the dataset since flwrlabs/pacs couldn't be found on the Hugging Face Hub
Found the latest cached dataset configuration 'default' at /home/apiasecz/.cache/huggingface/datasets/flwrlabs___pacs/default/0.0.0/394113073258ead631f617d2e13bb377c0715c4b (last modified on Wed Jan 29 16:59:35 2025).
Device: cuda
CUDA Version: 12.1
cuDNN Version: 8902
PyTorch Version: 2.3.0

Model used: PACSCNN
Initial dataset creation
Initial dataset creation
Initial loss: 3.672288179397583
<__main__.Policy object at 0x14b614b99d60>
3.672288179397583
Loss historical diff: 0.058788180351257324; Loss Difference: 0.0, Decision: 0
Round 1 took 4.781219005584717s, results:
{'t': 0, 'current_accuracy': 0.1796875, 'current_loss': 3.7310763597488403, 'train_loss': None, 'decision': 0, 'drift_rate': 0, 'target_domains': ['cartoon']}
Loss historical diff: 0.05878770351409912; Loss Difference: 0.0, Decision: 0
Round 2 took 4.256371974945068s, results:
{'t': 1, 'current_accuracy': 0.1796875, 'current_loss': 3.731075882911682, 'train_loss': None, 'decision': 0, 'drift_rate': 0, 'target_domains': ['cartoon']}
Loss historical diff: 0.05878782272338867; Loss Difference: 0.0, Decision: 0
Round 3 took 4.574238538742065s, results:
{'t': 2, 'current_accuracy': 0.1796875, 'current_loss': 3.7310760021209717, 'train_loss': None, 'decision': 0, 'drift_rate': 0, 'target_domains': ['cartoon']}
Loss historical diff: 0.058788180351257324; Loss Difference: 0.0, Decision: 0
Round 4 took 4.00834059715271s, results:
{'t': 3, 'current_accuracy': 0.1796875, 'current_loss': 3.7310763597488403, 'train_loss': None, 'decision': 0, 'drift_rate': 0, 'target_domains': ['cartoon']}
Loss historical diff: 0.05878782272338867; Loss Difference: 0.0, Decision: 0
Round 5 took 4.590805768966675s, results:
{'t': 4, 'current_accuracy': 0.1796875, 'current_loss': 3.7310760021209717, 'train_loss': None, 'decision': 0, 'drift_rate': 0, 'target_domains': ['cartoon']}
Loss historical diff: 0.058788299560546875; Loss Difference: 0.0, Decision: 0
Round 6 took 4.475101470947266s, results:
{'t': 5, 'current_accuracy': 0.1796875, 'current_loss': 3.73107647895813, 'train_loss': None, 'decision': 0, 'drift_rate': 0, 'target_domains': ['cartoon']}
Loss historical diff: 0.058788418769836426; Loss Difference: 0.0, Decision: 0
Round 7 took 5.320754528045654s, results:
{'t': 6, 'current_accuracy': 0.1796875, 'current_loss': 3.7310765981674194, 'train_loss': None, 'decision': 0, 'drift_rate': 0, 'target_domains': ['cartoon']}
Loss historical diff: 0.058788418769836426; Loss Difference: 0.0, Decision: 0
Round 8 took 4.4154884815216064s, results:
{'t': 7, 'current_accuracy': 0.1796875, 'current_loss': 3.7310765981674194, 'train_loss': None, 'decision': 0, 'drift_rate': 0, 'target_domains': ['cartoon']}
Loss historical diff: 0.05878758430480957; Loss Difference: 0.0, Decision: 0
Round 9 took 4.8184332847595215s, results:
{'t': 8, 'current_accuracy': 0.1796875, 'current_loss': 3.7310757637023926, 'train_loss': None, 'decision': 0, 'drift_rate': 0, 'target_domains': ['cartoon']}
Loss historical diff: 0.058788418769836426; Loss Difference: 0.0, Decision: 0
Round 10 took 4.362515211105347s, results:
{'t': 9, 'current_accuracy': 0.1796875, 'current_loss': 3.7310765981674194, 'train_loss': None, 'decision': 0, 'drift_rate': 0, 'target_domains': ['cartoon']}
Loss historical diff: 0.058788180351257324; Loss Difference: 0.0, Decision: 0
Round 11 took 4.3715293407440186s, results:
{'t': 10, 'current_accuracy': 0.1796875, 'current_loss': 3.7310763597488403, 'train_loss': None, 'decision': 0, 'drift_rate': 0.001, 'target_domains': ['photo']}
Loss historical diff: 0.05878794193267822; Loss Difference: 0.0, Decision: 0
Round 12 took 4.343749761581421s, results:
{'t': 11, 'current_accuracy': 0.1796875, 'current_loss': 3.7310761213302612, 'train_loss': None, 'decision': 0, 'drift_rate': 0.001670010459667194, 'target_domains': ['sketch']}
Loss historical diff: 0.058788180351257324; Loss Difference: 0.0, Decision: 0
Round 13 took 4.602599382400513s, results:
{'t': 12, 'current_accuracy': 0.1796875, 'current_loss': 3.7310763597488403, 'train_loss': None, 'decision': 0, 'drift_rate': 0.002338845493317048, 'target_domains': ['sketch']}
Loss historical diff: 0.05878782272338867; Loss Difference: 0.0, Decision: 0
Round 14 took 4.439319372177124s, results:
{'t': 13, 'current_accuracy': 0.1796875, 'current_loss': 3.7310760021209717, 'train_loss': None, 'decision': 0, 'drift_rate': 0.003005331737028868, 'target_domains': ['sketch']}
Loss historical diff: 0.05878770351409912; Loss Difference: 0.0, Decision: 0
Round 15 took 4.492081165313721s, results:
{'t': 14, 'current_accuracy': 0.1796875, 'current_loss': 3.731075882911682, 'train_loss': None, 'decision': 0, 'drift_rate': 0.003668299947457636, 'target_domains': ['sketch']}
Loss historical diff: 0.06016206741333008; Loss Difference: 0.0, Decision: 0
Round 16 took 4.220155954360962s, results:
{'t': 15, 'current_accuracy': 0.1796875, 'current_loss': 3.732450246810913, 'train_loss': None, 'decision': 0, 'drift_rate': 0.00432658705308415, 'target_domains': ['sketch']}
Loss historical diff: 0.04415082931518555; Loss Difference: 0.0, Decision: 0
Round 17 took 4.670241832733154s, results:
{'t': 16, 'current_accuracy': 0.1796875, 'current_loss': 3.7164390087127686, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0049790381946376765, 'target_domains': ['sketch']}
Loss historical diff: 0.03040468692779541; Loss Difference: 0.0, Decision: 0
Round 18 took 4.2579145431518555s, results:
{'t': 17, 'current_accuracy': 0.1796875, 'current_loss': 3.7026928663253784, 'train_loss': None, 'decision': 0, 'drift_rate': 0.005624508751111546, 'target_domains': ['sketch']}
Loss historical diff: 0.039947509765625; Loss Difference: 0.0, Decision: 0
Round 19 took 4.435744285583496s, results:
{'t': 18, 'current_accuracy': 0.1796875, 'current_loss': 3.712235689163208, 'train_loss': None, 'decision': 0, 'drift_rate': 0.006261866347817332, 'target_domains': ['sketch']}
Loss historical diff: 0.02398514747619629; Loss Difference: 0.0, Decision: 0
Round 20 took 4.748234748840332s, results:
{'t': 19, 'current_accuracy': 0.1796875, 'current_loss': 3.6962733268737793, 'train_loss': None, 'decision': 0, 'drift_rate': 0.006889992842954847, 'target_domains': ['sketch']}
Loss historical diff: 0.02312934398651123; Loss Difference: 0.0, Decision: 0
Round 21 took 4.148366212844849s, results:
{'t': 20, 'current_accuracy': 0.1796875, 'current_loss': 3.6954175233840942, 'train_loss': None, 'decision': 0, 'drift_rate': 0.007507786289212803, 'target_domains': ['sketch']}
Loss historical diff: 0.0; Loss Difference: 0.0, Decision: 0
Round 22 took 3.8177077770233154s, results:
{'t': 21, 'current_accuracy': 0.18359375, 'current_loss': 3.6694923639297485, 'train_loss': None, 'decision': 0, 'drift_rate': 0.00811416286695884, 'target_domains': ['sketch']}
Loss historical diff: 0.014425992965698242; Loss Difference: 0.0, Decision: 0
Round 23 took 4.5139076709747314s, results:
{'t': 22, 'current_accuracy': 0.1796875, 'current_loss': 3.6839183568954468, 'train_loss': None, 'decision': 0, 'drift_rate': 0.008708058785627446, 'target_domains': ['sketch']}
Loss historical diff: 0.026772618293762207; Loss Difference: 0.0, Decision: 0
Round 24 took 4.484662294387817s, results:
{'t': 23, 'current_accuracy': 0.18359375, 'current_loss': 3.6962649822235107, 'train_loss': None, 'decision': 0, 'drift_rate': 0.009288432149970084, 'target_domains': ['sketch']}
Loss historical diff: 0.02125370502471924; Loss Difference: 0.0, Decision: 0
Round 25 took 4.754352807998657s, results:
{'t': 24, 'current_accuracy': 0.18359375, 'current_loss': 3.6907460689544678, 'train_loss': None, 'decision': 0, 'drift_rate': 0.009854264787893505, 'target_domains': ['sketch']}
Loss historical diff: 0.05022561550140381; Loss Difference: 0.0, Decision: 0
Round 26 took 4.60505223274231s, results:
{'t': 25, 'current_accuracy': 0.1796875, 'current_loss': 3.7197179794311523, 'train_loss': None, 'decision': 0, 'drift_rate': 0.01040456403667957, 'target_domains': ['sketch']}
Loss historical diff: 0.024285554885864258; Loss Difference: 0.0, Decision: 0
Round 27 took 4.093108892440796s, results:
{'t': 26, 'current_accuracy': 0.1796875, 'current_loss': 3.693777918815613, 'train_loss': None, 'decision': 0, 'drift_rate': 0.010938364484452966, 'target_domains': ['sketch']}
Loss historical diff: 0.0448000431060791; Loss Difference: 0.0, Decision: 0
Round 28 took 4.232785224914551s, results:
{'t': 27, 'current_accuracy': 0.1796875, 'current_loss': 3.7142924070358276, 'train_loss': None, 'decision': 0, 'drift_rate': 0.011454729663841687, 'target_domains': ['sketch']}
Loss historical diff: 0.020777344703674316; Loss Difference: 0.0, Decision: 0
Round 29 took 4.37039852142334s, results:
{'t': 28, 'current_accuracy': 0.1796875, 'current_loss': 3.690269708633423, 'train_loss': None, 'decision': 0, 'drift_rate': 0.011952753694859018, 'target_domains': ['sketch']}
Loss historical diff: 0.04714393615722656; Loss Difference: 0.0, Decision: 0
Round 30 took 3.7669765949249268s, results:
{'t': 29, 'current_accuracy': 0.1796875, 'current_loss': 3.716636300086975, 'train_loss': None, 'decision': 0, 'drift_rate': 0.012431562874124853, 'target_domains': ['sketch']}
Loss historical diff: 0.09575343132019043; Loss Difference: 0.0, Decision: 1
/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/conv.py:456: UserWarning: Plan failed with an OutOfMemoryError: CUDA out of memory. Tried to allocate 1.00 GiB. GPU  (Triggered internally at /opt/conda/conda-bld/pytorch_1712608847532/work/aten/src/ATen/native/cudnn/Conv_v8.cpp:924.)
  return F.conv2d(input, weight, bias, self.stride,
/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/conv.py:456: UserWarning: Plan failed with an OutOfMemoryError: CUDA out of memory. Tried to allocate 514.00 MiB. GPU  (Triggered internally at /opt/conda/conda-bld/pytorch_1712608847532/work/aten/src/ATen/native/cudnn/Conv_v8.cpp:924.)
  return F.conv2d(input, weight, bias, self.stride,
/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/conv.py:456: UserWarning: Plan failed with an OutOfMemoryError: CUDA out of memory. Tried to allocate 258.00 MiB. GPU  (Triggered internally at /opt/conda/conda-bld/pytorch_1712608847532/work/aten/src/ATen/native/cudnn/Conv_v8.cpp:924.)
  return F.conv2d(input, weight, bias, self.stride,
/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/conv.py:456: UserWarning: Plan failed with an OutOfMemoryError: CUDA out of memory. Tried to allocate 516.00 MiB. GPU  (Triggered internally at /opt/conda/conda-bld/pytorch_1712608847532/work/aten/src/ATen/native/cudnn/Conv_v8.cpp:924.)
  return F.conv2d(input, weight, bias, self.stride,
/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/conv.py:456: UserWarning: Plan failed with an OutOfMemoryError: CUDA out of memory. Tried to allocate 770.00 MiB. GPU  (Triggered internally at /opt/conda/conda-bld/pytorch_1712608847532/work/aten/src/ATen/native/cudnn/Conv_v8.cpp:924.)
  return F.conv2d(input, weight, bias, self.stride,
Traceback (most recent call last):
  File "/home/apiasecz/programming/concept_drift_optimal_adaptation_FL/src/execute/evaluate_policy.py", line 753, in <module>
    main()
  File "/home/apiasecz/programming/concept_drift_optimal_adaptation_FL/src/execute/evaluate_policy.py", line 733, in main
    evaluate_policy_under_drift(
  File "/home/apiasecz/programming/concept_drift_optimal_adaptation_FL/src/execute/evaluate_policy.py", line 554, in evaluate_policy_under_drift
    update_loss = agent_train.update_steps(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/programming/Federated_Learning_Base_Toolkit_torch/fl_toolkit/federated/client.py", line 202, in update_steps
    return self.model.update_steps(self.data_loader, optimizer, loss_fn, num_updates, verbose)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/programming/Federated_Learning_Base_Toolkit_torch/fl_toolkit/models/base_model.py", line 93, in update_steps
    outputs = self.model(inputs)
              ^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/programming/concept_drift_optimal_adaptation_FL/src/execute/evaluate_policy.py", line 47, in forward
    x = self.features(x)
        ^^^^^^^^^^^^^^^^
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/container.py", line 217, in forward
    input = module(input)
            ^^^^^^^^^^^^^
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/programming/concept_drift_optimal_adaptation_FL/src/execute/evaluate_policy.py", line 70, in forward
    out = self.bn1(out)
          ^^^^^^^^^^^^^
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/batchnorm.py", line 175, in forward
    return F.batch_norm(
           ^^^^^^^^^^^^^
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/functional.py", line 2509, in batch_norm
    return torch.batch_norm(
           ^^^^^^^^^^^^^^^^^
torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 256.00 MiB. GPU 
